src/iop/sharpen.c

```diff
@@ -337,7 +337,7 @@ void process(struct dt_iop_module_t *self, dt_dev_pixelpipe_iop_t *piece, const
     const size_t start_row = j-rad;
     const size_t end_row = j+rad;
     // do the bulk of the row four at a time
-    for(int i = 0; i < width; i += 4)
+    for(int i = 0; i < (width & ~3); i += 4)
```

Comment:
This is a rather problematic kind of integer operations mixing signed and unsigned arithmetic of integer types with different width. Imho, the loop and the loops that follow should be refactored to use loop variables of type `size_t`. E.g.,:
```
for(size_t i = 0; i < (width & ~3u); i += 4u)
```
or even
```
for(size_t i = 0; i < (width & ~(size)3); i += (size)4)
```
when one wants to be very verbose. Note that `width` is of type `size_t`.